cpp/src/arrow/compute/exec/tpch_node.h

```diff
@@ -42,40 +42,23 @@ class ARROW_EXPORT TpchGen {
    * table from that single TpchGen instance. Note: Every batch will be scheduled as a new
    * task using the ExecPlan's scheduler.
    */
-  static Result<TpchGen> Make(ExecPlan* plan, double scale_factor = 1.0,
-                              int64_t batch_size = 4096,
-                              util::optional<int64_t> seed = util::nullopt);
+  static Result<std::unique_ptr<TpchGen>> Make(
```

Comment:
This does not need to be a unique_ptr. The whole point of using shared_ptr for the order/lineitem and part/partsupp generators is so that this TpchGen object can safely die. 
Also none of the methods in this class need to be virtual.

Reply:
As the commit title suggested, the point here is to hide implementation details from the `.h`, and minimize header inclusion.
It can be a `unique_ptr`, a `shared_ptr` or the pimpl pattern, but any of these patterns require some form of dynamic allocation.